### PR TITLE
refactor: Flatten messagelayer 

### PR DESCRIPTION
**Pull request by jkrvivian**
# Description of change

This PR copies files from the `binary/messagelayer` to `tangle/` with a flat structure and **does not** resolve the import issues from other packages or plugins, which would be done in the next PR.

This PR also includes: error messages annotation and using single character for receiver method.

## Type of change
- Enhancement (a non-breaking change which adds functionality)

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/tangle/data_payload.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+import (
+	"fmt"
+
+	"github.com/iotaledger/hive.go/marshalutil"
+	"github.com/iotaledger/hive.go/stringify"
+)
+
+const (
+	// ObjectName defines the name of the data object.
+	ObjectName = "data"
```
Comment (jonastheis) [inline packages/tangle/data_payload.go:12]: I think for now it's fine to name it `Object` but eventually we need to decide to stick to either `Object` or `Payload` and change it across the whole codebase. Even the file is called `data_payload.go`.

File: packages/tangle/approver.go [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+var _ objectstorage.StorableObject = &Approver{}
+
+// CachedApprover is a wrapper for a stored cached object representing an approver.
+type CachedApprover struct {
+	objectstorage.CachedObject
+}
+
+// Unwrap unwraps the cached approver into the underlying approver.
+// If stored object cannot be cast into an approver or has been deleted, it returns nil.
+func (cachedApprover *CachedApprover) Unwrap() *Approver {
```
Comment (jonastheis) [inline packages/tangle/approver.go:144]:
```suggestion
func (c *CachedApprover) Unwrap() *Approver {
```
or `a` would also be fine

**Review by jonastheis — CHANGES_REQUESTED** [review]

File: packages/tangle/message.go [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+	copy(result[:], bytes)
+
+	// return the number of bytes we processed
+	consumedBytes = MessageIDLength
+
+	return
+}
+
+// ParseMessageID is a wrapper for simplified unmarshaling in a byte stream using the marshalUtil package.
+func ParseMessageID(marshalUtil *marshalutil.MarshalUtil) (MessageID, error) {
```
[the remark this answers is not in the document]
Comment (jkrvivian) [reply packages/tangle/message.go:60]: okay, then I'll use the entity name as the prefix. :smile:

File: packages/tangle/requester.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+	scheduledRequestsMutex sync.RWMutex
+}
+
+// MessageExistsFunc is a function that tells if a message exists.
+type MessageExistsFunc func(messageId MessageID) bool
+
+// NewRequester creates a new message requester.
+func NewRequester(missingMessages []MessageID, optionalOptions ...Option) *Requester {
+	requester := &Requester{
```
[the remark this answers is not in the document]
Comment (jkrvivian) [reply packages/tangle/requester.go:56]: The single character is used for method receiver, yet `requester ` is a local variable, it's ok to stay like this I suppose.

**Review by jonastheis — APPROVED** [review]

File: packages/tangle/requester.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (hmoog) [reply packages/tangle/requester.go:56]: agree

**Review by capossele — APPROVED** [review]
Nice work!

**Review by hmoog — CHANGES_REQUESTED** [review]
I think those would be the last changes - rest LGTM

**Review by hmoog — APPROVED** [review]

**Pull request merged**